orchestrator/types.py

```diff
@@ -231,21 +232,18 @@ def is_union_type(t: Any, test_type: Optional[type] = None) -> bool:
     True
     >>> is_union_type(int)
     False
-
     """
-    if get_origin(t):
-        if get_origin(t) == Union:  # type: ignore
-            if test_type:
-                if is_of_type(t, test_type):
-                    return True
-                for arg in get_args(t):
-                    result = is_of_type(arg, test_type)
-                    if result:
-                        return result
-                return False
-            else:
-                return True
+    union_types = [Union, UnionType] if UnionType else [Union]
+    if not get_origin(t) or get_origin(t) not in union_types:
```

Comment:
I wonder if:

`if get_origin(t) not in union_types:`

is not sufficient. These multiple negations make it difficult to read.

Reply:
tried all scenarios in the doc-tests.